### PR TITLE
feat:Devise

### PR DESCRIPTION
**Pull request by taher-software**
In this branch, I've added the following changes:
- Install and setup devise.
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password

### CODE REVIEW

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<p>Welcome <%= @email %>!</p>
+
+<p>You can confirm your account email through the link below:</p>
+
+<p><%= link_to 'Confirm my account', user_confirmation_url(confirmation_token: @token) %></p>
```
Comment (dasileker) [inline app/views/devise/mailer/confirmation_instructions.html.erb:5]: - [ ] Greate job on implementing all the requirements, but I can't get the confirmation token to verify the inputs.

Capture web_1-3-2022_122837_localhost

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Taher-web-dev,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The PR is good and very descriptive.
- Linters are passing. 
- The Readme file describes all the steps.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/mailer/confirmation_instructions.html.erb (hunk 1 above)
Comment (taher-software) [reply app/views/devise/mailer/confirmation_instructions.html.erb:5]: @dasileker 
hi Amine ,
As I have showed you in our meeting , the app works smoothly in my system without any issue. So I will just ask for an other review as you had suggested me.

Comment (dasileker) [reply app/views/devise/mailer/confirmation_instructions.html.erb:5]: @Taher-web-dev It's weird that the app is functioning in your environment. I hope this problem is on my side.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Taher and Daniel,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 


<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**